### PR TITLE
queue: fix forwards ignoring group memberships

### LINKED ISSUE #465: Enabling/disabling groups have no effect on member forwards

### PR DESCRIPTION
**Pull request by AlbsAgren**
This commit fixes an issue where forwards would not receive configuration updates when their group memberships changed. This made it so that enabling or disabling groups that a forward was dependent on would not affect the forward.

Resolves #465

### CODE REVIEW

**Review by ahhud — APPROVED** [review]

File: middleware/queue/include/queue/forward/state.h [hunk 1]
```diff
@@ -446,7 +444,7 @@ namespace casual
 
          std::vector< std::string> memberships;
 
-         configuration::group::Coordinator group_coordinator;
+         bool enabled;
```
Comment (lazan) [inline middleware/queue/include/queue/forward/state.h:447]: Initialize the bool

Comment (lazan) [reply middleware/queue/include/queue/forward/state.h:447]: Can the whole forward group be enabled/disabled?

Comment (AlbsAgren) [reply middleware/queue/include/queue/forward/state.h:447]: Yes, which is asymmetric with gateway I suppose. I think forward was the first one we did.

Comment (lazan) [reply middleware/queue/include/queue/forward/state.h:447]: But shouldn't we shutdown the group? Or is to complicated?

Comment (AlbsAgren) [reply middleware/queue/include/queue/forward/state.h:447]: Yeah, I'm not sure why I did it like this. I imagine we had the same discussion back then, but I can't remember.

Comment (lazan) [reply middleware/queue/include/queue/forward/state.h:447]: I don't remember us talking about having a _disabled_ process running. I do remember us talking about keeping an _enable indicator_ for each forward state that the queue-manager has.

But then again... I don't really trust my memory :)

Comment (lazan) [reply middleware/queue/include/queue/forward/state.h:447]: I think you should change the samantics to shutdown a disabled group. You game?

Comment (AlbsAgren) [reply middleware/queue/include/queue/forward/state.h:447]: Aye

**Pull request closed without merging**